editoast/src/views/train_schedule/mod.rs

```diff
@@ -371,47 +374,58 @@ async fn get_result(
     Ok(Json(res))
 }
 
-#[derive(Deserialize, IntoParams)]
-struct GetResultsQuery {
+#[derive(Debug, Deserialize, ToSchema, Serialize)]
+struct TrainScheduleResults {
     /// The ID of the path that was used to project the train path
     path_id: Option<i64>,
-    /// The timetable ID
-    timetable_id: i64,
+    train_ids: Vec<i64>,
 }
 
 /// Retrieve the simulation result of multiple train schedules
 #[utoipa::path(
     tag = "train_schedule",
-    params(GetResultsQuery),
+    request_body = inline(TrainScheduleResults),
     responses(
         (status = 200, description = "The train schedule results", body = Vec<SimulationReport>)
     )
 )]
-#[get("/results")]
+#[post("/results")]
 async fn get_results(
     db_pool: Data<DbPool>,
-    query: Query<GetResultsQuery>,
+    request: Json<TrainScheduleResults>,
     core: Data<CoreClient>,
 ) -> Result<Json<Vec<SimulationReport>>> {
-    let query = query.into_inner();
+    use tables::train_schedule::dsl;
+    let body = request.into_inner();
+    let train_ids = body.train_ids;
+    let mut conn = db_pool.get().await?;
+    let schedules = dsl::train_schedule
+        .filter(dsl::id.eq_any(train_ids))
+        .load::<TrainSchedule>(&mut conn)
+        .await?;
+
+    if schedules.is_empty() {
+        return Ok(Json(vec![]));
+    }
+    let timetable_id = schedules[0].timetable_id;
+    if schedules.iter().any(|ts| ts.timetable_id != timetable_id) {
+        return Err(TrainScheduleError::BatchShouldHaveSameTimetable.into());
+    }
 
-    let timetable = Timetable::retrieve(db_pool.clone(), query.timetable_id)
+    let timetable = Timetable::retrieve(db_pool.clone(), timetable_id)
         .await?
-        .ok_or(TrainScheduleError::TimetableNotFound {
-            timetable_id: query.timetable_id,
-        })?;
+        .ok_or(TrainScheduleError::TimetableNotFound { timetable_id })?;
     let infra_version = timetable
         .infra_version_from_timetable(db_pool.clone())
         .await;
-    let schedules =
-        filter_invalid_trains(db_pool.clone(), timetable.id.unwrap(), infra_version).await?;
-    if schedules.is_empty() {
+    let result_schedules = filter_invalid_trains(db_pool.clone(), schedules, infra_version).await?;
+    if result_schedules.is_empty() {
```

Comment:
Maybe it would be best to return a type like

```
struct TrainSimulationResponse {
    simulations: Vec<(i64, SimulationReport)>,
    invalid_trains: Vec<i64>, // or Vec<(i64, TrainScheduleValidation)>
}
```

because at this point, returning a `vec![]` can be valid in different situations and have a completely different meaning.

Wdyt?